src/tests/quantuminspire/qiskit/test_backend_qx.py

```diff
@@ -300,89 +317,138 @@ def test_validate_nr_classical_qubits_less_than_nr_qubits_conditional_gate(self)
                                                    ' number of qubits when using conditional gate operations',
                                simulator.run, job)
 
-    def test_validate_operation_after_measure(self):
-        with patch.object(QuantumInspireBackend, "_submit_experiment", return_value=Mock()):
-            simulator = QuantumInspireBackend(Mock(), Mock())
-            instructions = [{'name': 'CX', 'qubits': [0, 1]},
-                            {'memory': [0], 'name': 'measure', 'qubits': [0]},
-                            {'name': 'X', 'qubits': [0]}]
-            job_dict = self._basic_qobj_dictionary
-            job_dict['experiments'][0]['instructions'] = instructions
-            job = qiskit.qobj.Qobj.from_dict(job_dict)
-            self.assertRaises(QisKitBackendError, simulator.run, job)
-
-    def test_no_operation_after_measure_cx_gate(self):
-        with patch.object(QuantumInspireBackend, "_submit_experiment", return_value=Mock()):
-            simulator = QuantumInspireBackend(Mock(), Mock())
-            instructions = [{'name': 'X', 'qubits': [1]},
+    def test_for_full_state_projection(self):
```

Comment:
Testing four cases in one unit test, should it not be four tests?

Reply:
I'll split them up